README.md

```diff
@@ -434,6 +441,45 @@ interface Loader {
 
   loadNow(fullSpecifier: string): void;
 }
+
+// ModuleInstance reifies an entangled pair of module environment record
+// and module exports namespace from a particular array of bindings
+// that correspond to the `import` and `export` declarations of a
+// module.
+interface ModuleInstance {
+  // Creates a module instance for either a static module record
+  // or a synthetic static module record.
+  // Creates a module environment record that defers to the
+  // global environment record of the designated loader.
+  // If the static module record needs a dynamic import behavior,
+  // creates a function that resolves import specifiers relative
+  // to its own referrer specifier and delegates to the
+  // attached loader.
+  constructor(
+    specifier: string,
+    record: StaticModuleRecord | SyntheticStaticModuleRecord,
+    loader: Loader,
+    importMeta?: Object,
```

Comment:
Threading a user-defined `import(fullSpecifier: string)` or removing the `specifier` argument and providing an `import(importSpecifier: string)` bound to the implied `fullSpecifier` might obviate one use of the `Loader`.

For hardened JavaScript isolation, we would need a way to construct a reified global environment record and pass that here instead of Loader.